de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeDefinitionTableLabelProvider.java

```diff
@@ -0,0 +1,116 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.requirements.ui.provider;
+
+import org.eclipse.emf.common.notify.AdapterFactory;
+import org.eclipse.swt.graphics.Image;
+
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ComposedPropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.util.PropertyInstanceValueSwitch;
+import de.dlr.sc.virsat.model.dvlm.general.IName;
+import de.dlr.sc.virsat.model.extension.requirements.model.EnumerationLiteral;
+import de.dlr.sc.virsat.model.extension.requirements.model.RequirementAttribute;
+import de.dlr.sc.virsat.model.ui.propertyinstance.util.PreferencedPropertyInstanceValueSwitchFactory;
+import de.dlr.sc.virsat.project.ui.labelProvider.VirSatTransactionalAdapterFactoryLabelProvider;
+
+/**
+ * @author fran_tb
+ *
+ */
+public class RequirementsAttributeDefinitionTableLabelProvider extends VirSatTransactionalAdapterFactoryLabelProvider {
+	
+	protected PropertyInstanceValueSwitch valueSwitch = PreferencedPropertyInstanceValueSwitchFactory.createInstance();
+	
+	/**
+	 * 
+	 * @param adapterFactory the adapter factory
+	 */
+	public RequirementsAttributeDefinitionTableLabelProvider(AdapterFactory adapterFactory) {
+		super(adapterFactory);
+		valueSwitch.setShowEnumValueDefinitionValues(false);
+	}
+	
+	@Override
+	public String getColumnText(Object object, int columnIndex) {
+		if (object == null) {
+			return "";
+		}
+		
+		if (object instanceof ComposedPropertyInstance) {
+			ComposedPropertyInstance cpi = (ComposedPropertyInstance) object;
+			CategoryAssignment ca = cpi.getTypeInstance();
+			redirectNotification(ca, object);
+			return getColumnText(cpi.getTypeInstance(), columnIndex);
+		}
+
+		if (object instanceof CategoryAssignment) {
+			CategoryAssignment ca = (CategoryAssignment) object;
+			
+			if (columnIndex == 0) {
+				return ca.getName();
+			} 
+			
+			switch (columnIndex) {
+				case 0:
+					return ca.getName();
+	
+				case 1:
+					return getAttributeTypeLabel(new RequirementAttribute(ca));
+					
+				case 2:
+					return getEnumerationDefinitionLabel(new RequirementAttribute(ca));
+					
+				default:
+					return null;
+			}
+		}
+		return ((IName) object).getName();
+	}
+	
+	/**
+	 * Show type value of the attribute
+	 * @param attDef the attribute definition
+	 * @return the type string
+	 */
+	protected String getAttributeTypeLabel(RequirementAttribute attDef) {
+		if (attDef == null || attDef.getType() == null) {
+			return "";
+		}
+		return attDef.getType();
+	}
+	
+	/**
+	 * Get the label of the contained enumeration
+	 * @param attDef the attribute
+	 * @return the label
+	 */
+	protected String getEnumerationDefinitionLabel(RequirementAttribute attDef) {

```

Comment:
This looks like a method that would be suitable for the model level so it can be tested.